static/app/views/settings/project/dynamicSampling/dynamicSampling.tsx

```diff
@@ -65,12 +65,15 @@ export function DynamicSampling({project}: Props) {
   async function handleToggle(type: DynamicSamplingBiasType) {
     addLoadingMessage();
 
-    const newDynamicSamplingBiases = biases.map(bias => {
-      if (bias.id === type) {
-        return {...bias, active: !bias.active};
-      }
-      return bias;
-    });
+    const biasIndex = biases.findIndex(b => b.id === type);
+
+    if (biasIndex === -1) {
+      return;
+    }
+
+    const newDynamicSamplingBiases = [...biases];
+    newDynamicSamplingBiases[biasIndex].active =
+      !newDynamicSamplingBiases[biasIndex].active;
```

Comment:
Isn't this mutating the original object?
I think `[...biases]` does not create a deep clone.
Perhaps your previous implementation with map and `{...bias}` was better in this regard?

Reply:
yes... I still need to manually test these changes, but I can use the `lodash/deepClone` here

Reply:
yeah, that would work, but honestly, I liked your map solution 👍

Reply:
yes... it was nice, but I needed to get the rule for this new logic https://github.com/getsentry/sentry/blob/0ad561f1c1ae61368ee1d92f00ec498f644f8da3/static/app/views/settings/project/dynamicSampling/dynamicSampling.tsx#L90 and I did not want to "loop" twice...but I have another idea and will update the code 😉